### ISSUE TITLE
Association always chooses primary key value regardless of foreign key column

### ISSUE THREAD

**Issue by nickhall** [thoughtbot/factory_girl#962]
I submitted this to Stack Overflow, but I'll repost it here as well. The original is formed as a question, so I'll state it simply here. On a table with a foreign key, FactoryGirl uses the value of the primary key of that table regardless of what column the foreign key points to. I'm on Rails 4.2.5.1.

It is important to note that I am using this with a legacy database and thus am not fully utilizing ActiveRecord, and I'm not using rails migrations at all.

http://stackoverflow.com/questions/40277018/how-to-make-factorygirl-foreign-key-association-point-to-proper-column

I have two tables: `notification` and `notification_type`. The `notification_type` table has an `ID` and a `code` column. The `notification` table has a `notitification_type` column with a foreign key on the `notificiation_type` table's `code` column.

    class Notification < ActiveRecord::Base
      ...
      belongs_to :notification_type, foreign_key: 'notification_type'
      ...

NotificationType:

    class NotificationType < ActiveRecord::Base
      has_many :notifications
    end

And then my factories:

    FactoryGirl.define do
      factory :notification_type do
        sequence(:name) { |n| "Test#{n}" }
        sequence(:code) { |n| n }
      end
    end

And the notification:

    FactoryGirl.define do
      factory :notification do
        notification_type
    
        trait :my_trait do
          topic
        end
      end
    end

When I call create on the notification model, it gives me a foreign key error because it's trying to create with the ID column of notification_type, not the code column. How can I make FactoryGirl build the object from the code column and not the ID column?

>ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  insert or update on table "notification" violates foreign key constraint "notification_notification_type_fkey"
DETAIL:  Key (notification_type)=(18) is not present in table "notification_type".

Where 18 is the ID, but I need it to use the code, which would be something like 1 or 2. I saw the `association_foreign_key` value, but it seems that it only works with the `has_and_belongs_to_many` association.

psql, just for closure:

>Foreign-key constraints:
    "notification_notification_type_fkey" FOREIGN KEY (notification_type) REFERENCES notification_type(code)

# tldr
The FactoryGirl association is using the primary key value despite the foreign key in the database pointing to a different column. How do I fix this?

**Comment by nickhall** [comment]
Closed because I've determined this is a rails issue, not FG.
